### ISSUE TITLE
Document the underlying data model for DataViews

### ISSUE THREAD

**Issue by mdickinson** [enthought/pyface#804]
The DataView documentation (https://docs.enthought.com/pyface/data_view.html) dives straight into indexing details after a brief overview, but doesn't explain what the (abstract) data model being indexed is. It's difficult to make sense of the indexing discussion without first having a mental model of the type of data that's being indexed; "hierarchical and non-hierarchical tabular data" isn't really enough here. Some extra words or explanation, or possibly even an example or two, would be really helpful.

**Comment by corranwebster** [comment]
I agree there could be more introductory explanation of the main parts of the DataView at the start.

There is, however, a fairly detailed discussion of the Data Model and how to implement one later on: https://docs.enthought.com/pyface/data_view.html#data-models

Are there parts of that that should be moved up, or is it more that i just needs a generic overview?

**Comment by mdickinson** [comment]
I'm looking for the overview necessary to start thinking about indexing. Basic things like:
- The basic structure is two-dimensional, and we think in terms of row and columns; a combination of row and column gives a "cell" containing a "value"
- Rows can have children
- Different rows at the same level can have different numbers of children
- Columns can't have children.

At the moment, all of this has to be guessed / inferred from subsequent discussion.
-
